spds/constraints.md

```diff
@@ -4,4 +4,71 @@ title: Constraints
 parent: SPD Language
 nav_order: 3
 permalink: /constraints/
----
\ No newline at end of file
+---
+# Constraints
+
+In SPD constraints restrict and alter the final behaviour of a policy. Constraints aid the modeller in 
+defining and analyzing policies that are more robust and less prone to oscillations.
+There are two types of constraints: temporal constraints and state based constraints. Furthermore, some constraints 
+are applicable for [ScalingPolicies](../docu/#spdScalingPolicy) and some for [TargetGroups](../docu/#targetsTargetGroup).
+
+## Temporal Constraints
+ 
+Temporal constraints are constraints in which the restriction and behavior alternation considers time as the main input source. 
+Two temporal constraints, which are applicable to a policy, are the [Interval Constraint](../docu/#constraintspolicyIntervalConstraint) and the [Cooldown Constraint](../docu/#constraintspolicyCooldownConstraint).
+One temporal constraint, that is applicable to the whole target group, is the [Thrashing Constraint](../docu/#constraintstargetThrashingConstraint).
+
+### Cooldown Constraint
+
+To exemplify the cooldown constraint, consider the image below which shows the behavior of a policy without a cooldown constraint.
+
+![example-constraint.png](..%2Fimages%2Fconstraints%2Fexample-constraint.png)
+
+The picture below depicts the behavior if a cooldown constraint is defined for a time period of 4 time units and with maximum scaling operation within that period of 0.
+Once the policy is enacted once and an adjustment is made there is no further adjustment for the next 4 time units.
+![example-cooldown-1.png](..%2Fimages%2Fconstraints%2Fexample-cooldown-1.png)
+
+In case the max number of scaling operations is set to 1, after the first adjustment, there is one additional operation allowed within the cooldown period but no more than one.
+![example-cooldown-2.png](..%2Fimages%2Fconstraints%2Fexample-cooldown-2.png)
+
+Cooldown constraints are applicable to the policy and help to allow the system to stabilize after another adjustment is made so that the effects are better observable.
+
+### Interval Constraint
+An interval constraint is defined by a time offset and a time period in which the policy is switched to an inactive state.
```

Comment:
i'd go for "[...] in which the policy is disregarded."
reasoning: inactive / active sound to me, like we are actually changing the `active` property in the model.